install_sct

```diff
@@ -757,3 +760,5 @@ else
     die "Installation validation Failed!"
   fi
 fi
+
+) 2>&1 | tee install_sct_log.txt
```

Comment:
Just to make sure I'm understanding correctly... from [What is the purpose of 'tee'?](https://superuser.com/a/1356843)

> `do_something | tee -a logfile` puts the output into logfile and to stdout, while `do_something >> logfile` puts it only into the logfile.
>
> The purpose of tee is to produce a one input, multiple output scenario - just like in a 'T' crossing.

So, we're using `tee` to print the output of `install_sct` to stdout _and_ the logfile. 

But... does this print output all at once, or does it print output bit by bit as the script runs in the subshell? Testing locally... it works as intended, so... I'm guessing I don't understand pipes as well as I thought I did, aha. :blush:

Reply:
Well it's sort of up to `tee` but as far as I know `tee` moves data as fast as it gets it.

It's possible a crash _could_ leave some data inside of `tee`'s internal buffers before getting flushed to the logfile/stdout, but the window is very short. Maybe a system under heavy load would be prone to that.